std/algorithm/comparison.d

```diff
@@ -901,10 +920,15 @@ template equal(alias pred = "a == b")
 {
     import std.meta : anySatisfy, allSatisfy;
 
+    enum isEmptyRange(R) = (isInputRange!R &&
```

Comment:
All input ranges can be checked for `empty`. What's the point of this?

Reply:
I added a comment explaining this.